eng/Analyzers.props

```diff
@@ -9,5 +9,6 @@
     <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="5.0.0-rtm.20502.2" PrivateAssets="all" />
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp.CodeStyle" Version="3.8.0-4.20503.2" PrivateAssets="all" />
     <PackageReference Include="StyleCop.Analyzers" Version="1.2.0-beta.205" PrivateAssets="all" />
+    <PackageReference Include="xunit.analyzers" Version="0.10.0" PrivateAssets="all" />
```

Comment:
Should we add an itemgroup that contains the test analyzers only? That way we don't add multiple `PackageReference` for not used analyzers into test projects?

Reply:
I wasn't sure how to structure that, as it seems we don't set the properties that declare whether something is a test project until after this has already been pulled in.  But maybe I've misunderstood the organization?

Reply:
I just tried it and it worked fine. I think it will work because the ItemGroups are evaluated after properties.

This is from the MSBuild docs:
>Because the items pass is after the properties pass, items can access any property if defined outside any targets, regardless of whether the property definition appears later.

So it should definitely work, that is why the `RunAnalyzers` condition in the `ItemGroup` works, that one is defined after this `.props` file is imported as well.

Reply:
Isn't xunit.analyzers already referenced by the xunit meta-package? https://github.com/dotnet/runtime/blob/88abe673eb28451764def48eee73972724d41dcc/eng/testing/xunit/xunit.props#L12

Reply:
> Isn't xunit.analyzers already referenced by the xunit meta-package?

Without this PackageReference, it doesn't work.

Reply:
I just checked, xunit.analyzers is a transitive dependency of the xunit package:

![image](https://user-images.githubusercontent.com/7412651/96281190-4aab6b00-0fd9-11eb-94ea-1d6bff9c04ee.png)

What exactly doesn't work?

Reply:
Yeah I looked at the assets.json of a test project and the XUnit.analyzers are there... perhaps we just need the ruleset?

Reply:
It was running the analyzers.  But I just tried removing it and it does appear to still be loaded.  So, must have been some other change I was experimenting with that prevented it from being used.